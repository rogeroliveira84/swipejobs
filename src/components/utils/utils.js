import moment from 'moment-timezone';

const Utils = {
    DateFormats: {
        full: 'ddd, MMM DD h:mm A z',
        short: 'ddd, MMM DD'
    },
    FormatDateTime: function (value, dateFormat) {
        if (!value || !dateFormat) {
            return null;
        }

        // todo: needs to implement a location based on job address, for now I put moment.tz.guess()
        const prepValue = value.replace(/\s+/g, '');          // because spaces are invalid (remove all spaces)
        const date = moment.tz(prepValue, moment.tz.guess()); // converts to local timezone 
        const dateString = date.format(dateFormat); // and apply format e.g: 'Mon, Sep 5 8:00 AM AEST'

        if (!date.isValid()) {
            return null;
        }

        return dateString;
    },
    FormatPeriod: function (periods, dateFormat) {
        if (periods && periods.length > 0) {
            const start = Utils.FormatDateTime(periods[0], dateFormat);
            const end = Utils.FormatDateTime(periods[periods.length - 1], dateFormat);
            return (start && end) ? `${start} - ${end}` : '';
        }
        return '';
    },
    FormatWage: value => `${((value ? Math.max(0, value) : 0) / 100).toFixed(2)}/hour`
}

export default Utils;
