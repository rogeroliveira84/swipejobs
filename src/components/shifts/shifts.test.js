import React from 'react';
import ReactDOM from 'react-dom';
import { dataMock } from '../../services/api/data';
import { shallow } from 'enzyme';
import utils from '../utils/utils';
import Shifts from './shifts';

describe('Shifts Component tests...', () => {
    
    const shiftsMock = dataMock.shifts.map(e => e.startDate);

    it('renders without crashing', () => {
        shallow(<Shifts dateFormat={utils.dateFormats.full} startDates={shiftsMock} />);
    });

    it('renders and matches the snapshot', () => {
        const component = shallow(<Shifts dateFormat={utils.dateFormats.full} startDates={shiftsMock} />);
        expect(component).toMatchSnapshot();
     });
  });