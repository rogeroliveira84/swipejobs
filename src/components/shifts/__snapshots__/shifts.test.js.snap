// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shifts Component tests... renders and matches the snapshot 1`] = `
Array [
  <div
    className="text-gray text-small bold date-part-row"
  >
    <DateParts
      dateString="Mon, Sep 05 7:00 AM AEST"
    />
  </div>,
  <div
    className="text-gray text-small bold date-part-row"
  >
    <DateParts
      dateString="Tue, Sep 06 7:00 AM AEST"
    />
  </div>,
  <div
    className="text-gray text-small bold date-part-row"
  >
    <DateParts
      dateString="Wed, Sep 07 7:00 AM AEST"
    />
  </div>,
  <div
    className="text-gray text-small bold date-part-row"
  >
    <DateParts
      dateString="Thu, Sep 08 7:00 AM AEST"
    />
  </div>,
  <div
    className="text-gray text-small bold date-part-row"
  >
    <DateParts
      dateString="Fri, Sep 09 7:00 AM AEST"
    />
  </div>,
  <div
    className="text-gray text-small bold date-part-row"
  >
    <DateParts
      dateString="Mon, Sep 12 7:00 AM AEST"
    />
  </div>,
  <div
    className="text-gray text-small bold date-part-row"
  >
    <DateParts
      dateString="Tue, Sep 13 7:00 AM AEST"
    />
  </div>,
  <div
    className="text-gray text-small bold date-part-row"
  >
    <DateParts
      dateString="Wed, Sep 14 7:00 AM AEST"
    />
  </div>,
  <div
    className="text-gray text-small bold date-part-row"
  >
    <DateParts
      dateString="Thu, Sep 15 7:00 AM AEST"
    />
  </div>,
  <div
    className="text-gray text-small bold date-part-row"
  >
    <DateParts
      dateString="Fri, Sep 16 7:00 AM AEST"
    />
  </div>,
]
`;
